src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -269,45 +473,58 @@ func (s *commitLogSource) ReadData(
 		// opportunity to repair the data instead of failing the bootstrap
 		// altogether.
 		s.log.Error("error in commitlog iterator", zap.Error(iterErr))
-		s.metrics.data.corruptCommitlogFile.Inc(1)
+		s.metrics.corruptCommitlogFile.Inc(1)
 		encounteredCorruptData = true
 	}
 
-	for _, encoderChan := range encoderChans {
-		close(encoderChan)
-	}
+	// Close the worker channels since we've enqueued all required data.
+	closeWorkerChannels()
 
 	// Block until all required data from the commit log has been read and
-	// encoded by the worker goroutines
+	// accumulated by the worker goroutines.
 	wg.Wait()
-	s.logEncodingOutcome(workerErrs, iter)
-
-	// Merge all the different encoders from the commit log that we created with
-	// the data that is available in the snapshot files.
-	s.log.Info("starting merge...")
-	mergeStart := time.Now()
-	bootstrapResult, err := s.mergeAllShardsCommitLogEncodersAndSnapshots(
-		ns,
-		shardsTimeRanges,
-		snapshotFilesByShard,
-		mostRecentCompleteSnapshotByBlockShard,
-		int(numShards),
-		blockSize,
-		shardDataByShard,
-	)
-	if err != nil {
-		return nil, err
+
+	// NB(r): Now that all accumulate workers are done we can release the
+	// series held onto by any checking out of series.
+	for _, elem := range namespaces.Namespaces.Iter() {
+		ns := elem.Value()
+		ns.DataAccumulator.Release()
 	}
-	s.log.Info("done merging...", zap.Duration("took", time.Since(mergeStart)))
 
+	// Log the outcome and calculate if required to return unfulfilled.
+	s.logAccumulateOutcome(workers, iter)
 	shouldReturnUnfulfilled, err := s.shouldReturnUnfulfilled(
-		encounteredCorruptData, ns, shardsTimeRanges, runOpts)
+		workers, encounteredCorruptData, initialTopologyState)
 	if err != nil {
-		return nil, err
+		return bootstrap.NamespaceResults{}, err
 	}
 
-	if shouldReturnUnfulfilled {
-		bootstrapResult.SetUnfulfilled(shardsTimeRanges)
+	bootstrapResult := bootstrap.NamespaceResults{
+		Results: bootstrap.NewNamespaceResultsMap(bootstrap.NamespaceResultsMapOptions{}),
```

Comment:
Worth initing this with a size? (Same comment whenever we call `NewNamespaceResultsMap` elsewhere)

Reply:
Opting to not, it's not in the hot path, once per bootstrap, and it's very low cardinality.